### PR TITLE
trafficserver: init at 9.0.0

### LINKED ISSUE #111750: Traffic Server
**Project description**
Traffic Server is a Apache product.

**Metadata**

* homepage URL: http://trafficserver.apache.org/
* source URL: https://github.com/apache/trafficserver
* license: mit, bsd, gpl2+, etc.
* platforms: unix, linux, darwin, etc.

### PR DESCRIPTION
**Pull request by heroku-miraheze**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Also include the `all-packages.nix` file modification.

**Comment by heroku-miraheze** [comment]
Included.

**Comment by AndersonTorres** [comment]
Squash the commits you did.

**Comment by heroku-miraheze** [comment]
Is not an easy, merge it now.

**Comment by fabaff** [comment]
Duplicate of #114090 ?

**Comment by heroku-miraheze** [comment]
No.

**Comment by heroku-miraheze** [comment]
I am going to close the duplicate now.

**Comment by heroku-miraheze** [comment]
Closed original.

**Comment by heroku-miraheze** [comment]
And merge it now.

**Comment by grahamc** [comment]
Sorry @heroku-miraheze, this attitude of demanding merges isn't how we operate here. If you'd like to have things merged right away, I'd encourage you to maintain your own fork of Nixpkgs. If you'd like to collaborate and cooporate with our community, then please behave accordingly.

**Comment by heroku-miraheze** [comment]
Done. When the last check is ended, merge it.

**Comment by AndersonTorres** [comment]
> Done. When the last check is ended, merge it.

Nope. The bot is not a final arbiter of code quality.

**Comment by heroku-miraheze** [comment]
But we need Traffic Server in NixPkgs.

**Comment by AndersonTorres** [comment]
It doesn't imply we are forced to commit a low-quality code merely because __we need trafficserver__.

**Comment by grahamc** [comment]
@AndersonTorres Don't worry about convincing, I don't think anyone with merge bit will be bullied in to merging poor quality work.

File: pkgs/servers/trafficserver/default.nix
Comment (dotlambda) [inline pkgs/servers/trafficserver/default.nix:41]:
```suggestion
    maintainers = with maintainers; [ joaquinito2051 ];
```
You need to add yourself to `maintainers/maintainer-list.nix`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
> * [x]  Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

at least not the bit about commit messages

**Comment by heroku-miraheze** [comment]
Updated all, now, when the last check is ended, merge it.

**Comment by adisbladis** [comment]
> Updated all, now, when the last check is ended, merge it.

Stop demanding your changes getting merged.

If you want things merged collaborate with the people reviewing your PR instead. 
That's the only way forward.

**Comment by heroku-miraheze** [comment]
I am collaborating with people.

**Comment by adisbladis** [comment]
You obviously still haven't read https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md which states how commit messages should look like.

**Comment by heroku-miraheze** [comment]
I cannot review collaborators, because i am from Windows.

**Comment by dotlambda** [comment]
> I cannot review collaborators, because i am from Windows.

It does not make sense for you to contribute if you cannot even test your own pull requests.

**Comment by heroku-miraheze** [comment]
Hello again, i need to review contributors, but i am from Windows, and you need to wait until install NixOS.

**Comment by heroku-miraheze** [comment]
Installed, wait...

**Comment by heroku-miraheze** [comment]
##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on ARCHITECTURE
- [x] executables tested on ARCHITECTURE
- [x] all depending packages build

##### Possible improvements

##### Comments

**Comment by heroku-miraheze** [comment]
https://gist.github.com/heroku-miraheze/e0f485bbe071289a9ba8199f104ab3a3

**Review by etu — CHANGES_REQUESTED** [review]
I probably would merge this code in this state if it wasn't for the bad commit history.

The commit history doesn't follow the https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md guidelines.

**Review by heroku-miraheze — COMMENTED** [review]
Reviewed

**Comment by dotlambda** [comment]
> * [x]  executables tested on ARCHITECTURE

Which distro did you use? I get
```
/nix/store/dgdzy2hp8iw3h9k3i04j4plb0ylmy4m7-trafficserver-9.0.0/bin/trafficserver: line 108: sed: command not found
This script needs to be ported to this OS
```
It seems like this software will be of no use on NixOS without upstream support.

**Comment by heroku-miraheze** [comment]
I use Ubuntu.

**Comment by AndersonTorres** [comment]
The `trafficserver` is a shell script that uses a lot of that "classic elegant hackery" and many assumptions about FHS distros.

It would be really cool if we ported it to NixOS.

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25466]:
```suggestion
  trafficserver = callPackage ../servers/trafficserver { };
```

File: pkgs/servers/trafficserver/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/trafficserver/default.nix:19]: Those belong into buildInputs and autoconf should pick them up then.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Kreyren** [comment]
> You obviously still haven't read https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md which states how commit messages should look like.

FWIW based on my interactions with him on matrix he's not english speaker and there is a language barrier for him to understand those guidelines.

**Comment by khrj** [comment]
> this attitude of demanding merges isn't how we operate here

(adding on to the previous comment) The author probably didn't intend to be demanding, but just came off that way because of whatever translation service they're using

**Pull request closed without merging**
